tests/integration/cli/commands/test_local_project_lock.py

```diff
@@ -0,0 +1,133 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import os
+import re
+import shutil
+import subprocess
+
+import pytest
+
+from pex.cli.testing import run_pex3
+from pex.common import touch
+from pex.testing import IS_PYPY, PY_VER, run_pex_command
+from pex.typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import Any, Callable
+
+
+@pytest.fixture
+def ansicolors_1_1_8(clone):
+    # type: (Callable[[str, str], str]) -> str
+    return clone(
+        "https://github.com/jonathaneunice/colors", "c965f5b9103c5bd32a1572adb8024ebe83278fb0"
+    )
+
+
+def test_fingerprint_stability(
+    tmpdir,  # type: Any
+    ansicolors_1_1_8,  # type: str
+):
+    # type: (...) -> None
+
+    lock = os.path.join(str(tmpdir), "lock")
+    pex_root = os.path.join(str(tmpdir), "pex_root")
+    run_pex3(
+        "lock", "create", "--pex-root", pex_root, ansicolors_1_1_8, "-o", lock
+    ).assert_success()
+
+    print_colors_version_args = [
+        "--pex-root",
+        pex_root,
+        "--lock",
+        lock,
+        "--",
+        "-c",
+        "import colors; print(colors.__version__)",
+    ]
+    result = run_pex_command(args=print_colors_version_args)
+    result.assert_success()
+    assert "1.1.8" == result.output.strip()
+
+    # Running the test suite generates .pyc files which should not count against the project
```

Comment:
@benjyw thanks for all the prior reviews. All that work was to support this stability bit. That required building the local project into an sdist and hashing the sdist instead of hashing the local project dir directly. That, in turn, required resolving build system requirements if the local project was PEP-517/PEP-518, and that led to circularity issues with the standard resolver which led to all the circularity refactor PRs.